src/components/settings/EditOrganizationInformationsDialog.tsx

```diff
@@ -0,0 +1,323 @@
+import CountryCodes from '~/public/countryCode.json'
+
+const FILE_MAX_SIZE = 800000
+
+const countryData: { value: string; label: string }[] = Object.keys(CountryCodes).map(
+  (countryKey) => {
+    return {
+      value: countryKey,
+      // @ts-ignore
+      label: CountryCodes[countryKey],
+    }
+  }
+)
+
+gql`
+  fragment EditOrganizationInformationsDialog on Organization {
+    id
+    logoUrl
+    name
+    legalName
+    legalNumber
+    email
+    addressLine1
+    addressLine2
+    zipcode
+    city
+    state
+    country
+  }
+
+  mutation updateOrganizationInformations($input: UpdateOrganizationInput!) {
+    updateOrganization(input: $input) {
+      ...OrganizationInformations
+      ...EditOrganizationInformationsDialog
+    }
+  }
+  ${OrganizationInformationsFragmentDoc}
+`
+
+export interface EditOrganizationInformationsDialogRef extends DialogRef {}
+
+interface EditOrganizationInformationsProps {
+  organization: EditOrganizationInformationsDialogFragment
+}
+
+export const EditOrganizationInformationsDialog = forwardRef<
+  DialogRef,
+  EditOrganizationInformationsProps
+>(({ organization }: EditOrganizationInformationsProps, ref) => {
+  const { translate } = useInternationalization()
+  const mounted = useRef(false)
+  const [updateOrganizationInformations] = useUpdateOrganizationInformationsMutation({
+    context: { silentErrorCodes: [Lago_Api_Error.UnprocessableEntity] },
+  })
+  const [logo, setLogo] = useState<string>()
+  const [logoUploadError, setLogoUploadError] = useState(false)
+  const hiddenFileInputRef = useRef<HTMLInputElement>(null)
+  const formikProps = useFormik<UpdateOrganizationInput>({
+    initialValues: {
+      legalName: organization?.legalName || '',
+      legalNumber: organization?.legalNumber || '',
+      email: organization?.email || '',
+      addressLine1: organization?.addressLine1 || '',
+      addressLine2: organization?.addressLine2 || '',
+      zipcode: organization?.zipcode || '',
+      city: organization?.city || '',
+      state: organization?.state || '',
+      country: organization?.country || undefined,
+    },
+    validationSchema: object().shape({
+      email: string().email('text_620bc4d4269a55014d493fc3'),
+    }),
+    enableReinitialize: true,
+    validateOnMount: true,
+    onSubmit: async ({ ...values }, formikBag) => {
+      await updateOrganizationInformations({
+        variables: {
+          input: {
+            logo,
+            ...values,
+          },
+        },
+        refetchQueries: ['getOrganizationInformations'],
+      })
+      ;(ref as unknown as RefObject<DialogRef>)?.current?.closeDialog()
+      if (mounted.current) {
+        formikBag.resetForm()
+      }
+    },
+  })
+
+  const getBase64 = (file: Blob) => {
+    var reader = new FileReader()
+
+    if (file.size > FILE_MAX_SIZE) {
+      setLogoUploadError(true)
+      return
+    }
+
+    reader.readAsDataURL(file)
+    reader.onload = function () {
```

Comment:
Prefer arrow function for more consistency in the code please

Reply:
Sure, works with arrow finally